### PR TITLE
[Feature Store] Move feature enrichment/imputing into `get_online_features`

### PR DESCRIPTION
**Pull request by yaronha**

### CODE REVIEW

File: docs/feature-store/feature-vectors.md [hunk 1]
```diff
@@ -140,4 +149,9 @@ entities = [{<feature-vector-entity-column-name>: <entity>}]
 svc.get(entities)
 ```
 
+The `entities` can be a list of dictionaries as shown in the example, or a list of list where the values in the internal 
+list correspond to the entity values (e.g. `entities = [["Joe"], ["Mike"]]`). the `.get()` method returns a dict by default
```
Comment (dinal) [inline docs/feature-store/feature-vectors.md:153]: this doesn't work. it has to a list of dict

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -162,13 +163,25 @@ def get_online_feature_service(
...
+    example with imputing::
+
+        svc = get_online_feature_service(vector_uri, impute_policy={"*": "$mean", "amount": 0))
+        resp = svc.get([{"id": "C123487"}])
+
     :param feature_vector:    feature vector uri or FeatureVector object
     :param run_config:        function and/or run configuration for remote jobs/services
+    :param impute_policy:     a dict with `impute_policy` per feature, the dict key is the feature name and the dict
+                              value indicate which value will be used in case the feature is NaN/empty, the replaced
+                              value can be fixed number for constants or $mean, $max, etc. for statistical values.
```
Comment (dinal) [inline mlrun/feature_store/api.py:175]: better state exactly what other options are there

**Review by dinal — APPROVED** [review]

File: mlrun/feature_store/feature_vector.py [hunk 3]
```diff
@@ -332,26 +333,105 @@ def add_feature(name, alias, feature_set_object):
...
+    def __init__(self, vector, graph, index_columns, impute_policy: dict = None):
         self.vector = vector
+        self.impute_policy = impute_policy or {}
+
         self._controller = graph.controller
         self._index_columns = index_columns
+        self._impute_values = {}
+
+    def load(self):
+        """internal, init the feature service and prep the imputing logic"""
```
Comment (Hedingber) [inline mlrun/feature_store/feature_vector.py:345]: so `initialize` no ?

File: mlrun/feature_store/feature_vector.py [hunk 4]
```diff
@@ -332,26 +333,105 @@ def add_feature(name, alias, feature_set_object):
...
     def status(self):
         """vector merger function status (ready, running, error)"""
         return "ready"
 
-    def get(self, entity_rows: List[dict], as_list=False):
-        """get feature vector given the provided entity inputs"""
+    def get(self, entity_rows: List[Union[dict, list]], as_list=False):
+        """get feature vector given the provided entity inputs
+
+        take a list of input vectors/rows and return a list of eriched feature vectors
```
Comment (Hedingber) [inline mlrun/feature_store/feature_vector.py:386]:
```suggestion
        take a list of input vectors/rows and return a list of enriched feature vectors
```

File: mlrun/feature_store/feature_vector.py [hunk 5]
```diff
@@ -332,26 +333,105 @@ def add_feature(name, alias, feature_set_object):
...
+
+        # validate we have valid input struct
+        if (
+            not entity_rows
+            or not isinstance(entity_rows, list)
+            or not isinstance(entity_rows[0], (list, dict))
+        ):
             raise mlrun.errors.MLRunInvalidArgumentError(
-                f"{entity_rows} is of type {type(entity_rows)}. It should be list of Dictionaries"
+                f"input data is of type {type(entity_rows)}. must be a list of list or list of dict"
```
Comment (Hedingber) [inline mlrun/feature_store/feature_vector.py:418]:
```suggestion
                f"input data is of type {type(entity_rows)}. must be a list of lists or list of dicts"
```

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/feature_vector.py (hunk 3 above)
Comment (yaronha) [reply mlrun/feature_store/feature_vector.py:345]: ok `init`

Comment (yaronha) [reply mlrun/feature_store/feature_vector.py:345]: changed to `initialize `

**Pull request merged**